### DISCUSSION TITLE
Can Fooocus be used on a computer without a GPU?

### DISCUSSION THREAD

**Discussion by poseny** [lllyasviel/Fooocus#1775] [Q&A] [answered]
I'm trying to install Fooocus on a computer without a GPU, I saw [here](https://github.com/lllyasviel/Fooocus?tab=readme-ov-file#minimal-requirement) that it's possible
The problem is that it writes me ERROR, and in CMD it returns me the error
RuntimeError: Found no NVIDIA driver on your system. Please check that you have an NVIDIA GPU and installed a driver from http://www.nvidia.com/Download/index.aspx
Is there a way to solve this?

**Comment by gaspain** [comment, marked as answer]
Yes. you must add --always-cpu to the run.bat file. Make it like this:
`.\python_embeded\python.exe -s Fooocus\entry_with_update.py --always-cpu
pause
`
It is 1000x slower. 1 image will take around 1hour on a modern high end laptop. And you need at least 32gb ram if not more. It will take a long time to even load the program as well.

**Reply by mianYaseen** [reply]
![image](https://github.com/user-attachments/assets/2f252483-5a63-4298-b8cf-3371e6e0781d)

bro not working for me in DeFooocus

**Reply by mashb1t** [reply]
@mianYaseen this is not DeFooocus, please open an issue there. Most likely your swap is not correctly set up or has insufficient capacity.

**Reply by MAVD84** [reply]
> Yes. you must add --always-cpu to the run.bat file. Make it like this: `.\python_embeded\python.exe -s Fooocus\entry_with_update.py --always-cpu pause ` It is 1000x slower. 1 image will take around 1hour on a modern high end laptop. And you need at least 32gb ram if not more. It will take a long time to even load the program as well.

Thank you 🙏

**Reply by ugintl** [reply]
Worked for me. `pause` should be in new line.

**Comment by raulpony** [comment]
Hi, i changed my run.bat file to this: 
`.\python_embeded\python.exe -s Fooocus\entry_with_update.py --always-cpu pause`

but it just loads up the cmd and then closes immediatly without opening the browser page, any suggestions on how to fix it?

**Reply by mashb1t** [reply]
@raulpony make sure pause is on a separate line after the entrypoint call 

```
.\python_embeded\python.exe -s Fooocus\entry_with_update.py --always-cpu
pause
```

**Reply by MAVD84** [reply]
It works for me, thanks!

**Reply by MGcrazen** [reply]
![image](https://github.com/user-attachments/assets/a2cd4d1c-319f-4128-8fea-164c3e5afd0c)
Doesn't work for me :(

**Reply by mashb1t** [reply]
@MGcrazen you can fix your spelling and try again

**Comment by MAVD84** [comment]
Can i use The Radeon R5 430 ?

**Reply by mashb1t** [reply]
No, it has 2GB VRAM and min. requirement is 8GB for AMD, see minimal system requirements in readme.

**Reply by MAVD84** [reply]
Thanks

**Reply by rmspot** [reply]
minimal requirements state 8GB but also that "If your device capability is lower than this spec, you **_may_** not be able to use Fooocus locally" -> Does it mean that we can try, and in the best case it works even if slower?
Or the minimum 8GB is strictly checked by the sw?
(by the way; I have a Nvidia M600M with 2GB ...)

**Reply by MAVD84** [reply]
Thank you

**Comment by maryf-t-s** [comment]


**Reply by mashb1t** [reply]
@maryf-t-s please find the system requirements for cpu only in the readme. It's not recommended as speed is abysmal though compared to GPU.

**Comment by maryf-t-s** [comment]
@mashb1t Thank you for your response! , I already know that my computer is too weak, but the thing is I'm trying to figure something out because I need to buy one that can handle Fooocus properly. What computer would be able to manage it? Please give me suggestions starting from the lowest price. Thank you in advance for any suggestions and answers!

-----------------From this description, it's hard for me to choose a laptop that would handle it. Maybe someone has a recommendation?
---------------------
Below is a test on a relatively low-end laptop with 16GB System RAM and 6GB VRAM (Nvidia 3060 laptop). The speed on this machine is about 1.35 seconds per iteration. Pretty impressive – nowadays laptops with 3060 are usually at very acceptable price.

image

Besides, recently many other software report that Nvidia driver above 532 is sometimes 10x slower than Nvidia driver 531. If your generation time is very long, consider download [Nvidia Driver 531 Laptop](https://www.nvidia.com/download/driverResults.aspx/199991/en-us/) or [Nvidia Driver 531 Desktop](https://www.nvidia.com/download/driverResults.aspx/199990/en-us/).
